packages/openapi-v3/src/parameter-decorator.ts

```diff
@@ -105,376 +102,268 @@ export namespace param {
     /**
      * Define a parameter of "dateTime" type that's read from the query string.
-     * Usage: @param.query.dateTime('paramName')
+     * "Usage": @param.query.dateTime('paramName')
      *
      * @param name Parameter name.
      */
-    dateTime: createParamShortCut(
-      'query',
-      typeAndFormatMap.dateTime.type,
-      typeAndFormatMap.dateTime.format,
-    ),
+    dateTime: createParamShortcut('query', typeAndFormatMap.dateTime),
     /**
      * Define a parameter of "password" type that's read from the query string.
-     * Usage: @param.query.password('paramName')
+     * "Usage": @param.query.password('paramName')
      *
      * @param name Parameter name.
      */
-    password: createParamShortCut(
-      'query',
-      typeAndFormatMap.password.type,
-      typeAndFormatMap.password.format,
-    ),
+    password: createParamShortcut('query', typeAndFormatMap.password),
   };
 
   export const header = {
     /**
      * Define a parameter of "string" type that's read from a request header.
-     * Usage: @param.header.string('paramName')
+     * "Usage": @param.header.string('paramName')
      *
      * @param name Parameter name, it must match the header name
      * (e.g. `Content-Type`).
      */
-    string: createParamShortCut('header', 'string'),
+    string: createParamShortcut('header', typeAndFormatMap.string),
     /**
      * Define a parameter of "number" type that's read from a request header.
-     *  Usage: @param.header.number('paramName')
+     *  "Usage": @param.header.number('paramName')
      *
      * @param name Parameter name, it must match the header name
      * (e.g. `Content-Length`).
      */
-    number: createParamShortCut('header', 'number'),
+    number: createParamShortcut('header', typeAndFormatMap.number),
     /**
      * Define a parameter of "boolean" type that's read from a request header.
-     *  Usage: @param.header.boolean('paramName')
+     *  "Usage": @param.header.boolean('paramName')
      *
      * @param name Parameter name, it must match the header name
      * (e.g. `DNT` or `X-Do-Not-Track`).
      */
-    boolean: createParamShortCut('header', 'boolean'),
+    boolean: createParamShortcut('header', typeAndFormatMap.boolean),
     /**
      * Define a parameter of "integer" type that's read from a request header.
-     *  Usage: @param.header.integer('paramName')
+     *  "Usage": @param.header.integer('paramName')
      *
      * @param name Parameter name, it must match the header name
      * (e.g. `Content-Length`).
      */
-    integer: createParamShortCut(
-      'header',
-      typeAndFormatMap.integer.type,
-      typeAndFormatMap.integer.format,
-    ),
+    integer: createParamShortcut('header', typeAndFormatMap.integer),
     /**
      * Define a parameter of "long" type that's read from a request header.
-     *  Usage: @param.header.long('paramName')
+     *  "Usage": @param.header.long('paramName')
```

Comment:
I think you misunderstood my comment. The code needs to be put in backticks to tell the markdown renderer to use a different font and space handling.

I tried to run `lerna run build:apidocs`, this command fails in openapi-v2 package. Then I tried to run `lb-apidocs` manually from openapi-v3, which works as expected, but prints errors about missing type definitions of `lodash` - I fixed them via `npm install --save @types/lodash`.

The apidocs generated by this command contain only like three entries, the param decorators are not included for example. The next step is to fix the list of source files in `docs.json`, see https://github.com/strongloop/loopback-next/blob/fe119f0829b9a3549b586fd8c2af6f5254ed16c5/packages/openapi-v3/docs.json#L2

The new content I used:

```json
{
  "content": [
    "src/api-decorator.ts",
    "src/controller-spec.ts",
    "src/generate-schema.ts",
    "src/index.ts",
    "src/json-to-schema.ts",
    "src/keys.ts",
    "src/operation-decorator.ts",
    "src/parameter-decorator.ts",
    "src/request-body-decorator.ts",
    "src/spec-builder"
  ],
  "codeSectionDepth": 4
}
```

With this change in place, the param decorators are still not documented - I think this is a bug of strong-docs? cc @raymondfeng 

So, I wanted to show how your tsdoc comments are rendered to point out what I expect to be a problem, but it's not possible. I guess you will have to trust me here :)

At minimum, please use the following syntax (no double-quotes around the word _Usage_, backticks around the code snippet):

```
* Usage: `@param.header.long('paramName')`
```

Also please fix `docs.json`, run `lb-apidocs` yourself, `open apidocs/index.html` and fix any problems you can spot. For example, do we want `paramShortcutOptions` and `TypeAndFormat` to show in the generated docs? I think these are private and should not be exposed to framework users. (Hint: use `@private` annotation.)

Reply:
> I guess you will have to trust me here :)

lol definitely...last time Kevin also points out some jsdoc bugs in my PR, was about to verify them but none of the docs work on http://apidocs.loopback.io/ 
Is it a known bug or we haven't prepared or it should work? If it should work I can file an issue, maybe not MVP but I want to fix it asap.

Reply:
@bajtos I run `npm run build:apidocs` in `openapi-v3` package but it doesn't print out error... 

I add `@private` to those functions for internal use, but jsdoc still display them, probably the fix need to be in jsdoc repo.
`parseRequestUrl` in `rest` package is also annotated with `@private`, and still shows up in jsdoc page.